be/src/olap/types.h

```diff
@@ -1193,6 +1208,23 @@ struct TypeTraits : public FieldTypeTraits<field_type> {
     static const int32_t size = sizeof(CppType);
 };
 
+// get type at compile time for performance
+// ---- Version_1 ----
+// template <FieldType field_type>
+// extern TypeInfo* get_scalar_type_info_by_template(){
+//     static constexpr TypeTraits<field_type> traits;
+//     static auto obj = ScalarTypeInfo(traits);
+//     static constexpr ScalarTypeInfo* p = &obj;
+//     return (TypeInfo*)(p);
+// }
+
+// Version_2
+template <FieldType field_type>
+std::shared_ptr<const TypeInfo> get_scalar_type_info_by_template() {
+    static constexpr TypeTraits<field_type> traits;
+    return std::shared_ptr<const TypeInfo>(new ScalarTypeInfo(traits));
```

Comment:
how about define static TypeInfo object and return reference. now， it seems we need to
create ScalarTypeInfo every function call